forgit.plugin.zsh

```diff
@@ -1,511 +1,112 @@
-hash emojify &>/dev/null && forgit_emojify='|emojify'
-
-# extract the first git sha occurring in the input and strip trailing newline
-forgit_extract_sha="grep -Eo '[a-f0-9]+' | head -1 | tr -d '[:space:]'"
-
-forgit_pager=${FORGIT_PAGER:-$(git config core.pager || echo 'cat')}
-forgit_show_pager=${FORGIT_SHOW_PAGER:-$(git config pager.show || echo "$forgit_pager")}
-forgit_diff_pager=${FORGIT_DIFF_PAGER:-$(git config pager.diff || echo "$forgit_pager")}
-forgit_ignore_pager=${FORGIT_IGNORE_PAGER:-$(hash bat &>/dev/null && echo 'bat -l gitignore --color=always' || echo 'cat')}
-forgit_blame_pager=${FORGIT_BLAME_PAGER:-$(git config pager.blame || echo "$forgit_pager")}
-forgit_enter_pager=${FORGIT_ENTER_PAGER:-"LESS='-r' less"}
+forgit::warn() { printf "%b[Warn]%b %s\n" '\e[0;33m' '\e[0m' "$@" >&2; }
 
-forgit_log_format=${FORGIT_LOG_FORMAT:-%C(auto)%h%d %s %C(black)%C(bold)%cr%Creset}
-forgit_log_preview_options="--graph --pretty=format:'$forgit_log_format' --color=always --abbrev-commit --date=relative"
-forgit_fullscreen_context=${FORGIT_FULLSCREEN_CONTEXT:-10}
-forgit_preview_context=${FORGIT_PREVIEW_CONTEXT:-3}
+# set installation path
+if [[ -n "$ZSH_VERSION" ]]; then
+    # shellcheck disable=2277,2296,2299
+    0="${ZERO:-${${0:#$ZSH_ARGZERO}:-${(%):-%N}}}"
+    # shellcheck disable=2277,2296,2298
+    0="${${(M)0:#/*}:-$PWD/$0}"
+    FORGIT_INSTALL_DIR="${0:h}"
+elif [[ -n "$BASH_VERSION" ]]; then
+    FORGIT_INSTALL_DIR="$(dirname -- "${BASH_SOURCE[0]}")"
+else
+    forgit::warn "Only zsh and bash are fully supported"
+fi
+if [[ -n "$FORGIT_INSTALL_DIR" ]]; then
+    export FORGIT_INSTALL_DIR
+fi
 
-# git commit viewer
 forgit::log() {
-    forgit::inside_work_tree || return 1
-    local opts graph files log_format preview_cmd enter_cmd
-    files=$(sed -nE 's/.* -- (.*)/\1/p' <<< "$*") # extract files parameters for `git show` command
-    preview_cmd="echo {} | $forgit_extract_sha | xargs -I% git show --color=always -U$forgit_preview_context % -- $files | $forgit_show_pager"
-    enter_cmd="echo {} | $forgit_extract_sha | xargs -I% git show --color=always -U$forgit_fullscreen_context % -- $files | $forgit_show_pager"
-    opts="
-        $FORGIT_FZF_DEFAULT_OPTS
-        +s +m --tiebreak=index
-        --bind=\"enter:execute($enter_cmd | $forgit_enter_pager)\"
-        --bind=\"ctrl-y:execute-silent(echo {} | $forgit_extract_sha | ${FORGIT_COPY_CMD:-pbcopy})\"
-        --preview=\"$preview_cmd\"
-        $FORGIT_LOG_FZF_OPTS
-    "
-    graph=--graph
-    [[ $FORGIT_LOG_GRAPH_ENABLE == false ]] && graph=
-    log_format=${FORGIT_GLO_FORMAT:-$forgit_log_format}
-    eval "git log $graph --color=always --format='$log_format' $* $forgit_emojify" |
-        FZF_DEFAULT_OPTS="$opts" fzf
-    fzf_exit_code=$?
-    # exit successfully on 130 (ctrl-c/esc)
-    [[ $fzf_exit_code == 130 ]] && return 0
-    return $fzf_exit_code
+    "$FORGIT_INSTALL_DIR/bin/git-forgit" log "$@"
```

Comment:
Give `"$FORGIT_INSTALL_DIR/bin/git-forgit"` a name like `$FORGIT_BIN`?

Reply:
That's a legit suggestion, and I even already did this within `git-forgit`, so it totally makes sense to also do it in the plugin. I will add a patch.